app/utils/injectSaga.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import getInjectors from './sagaInjectors';
+
+/**
+ * Dynamically injects a saga, passes component's props as saga arguments
+ *
+ * @param {string} key A key of the saga
+ * @param {function} saga A root saga that will be injected
+ * @param {string} [mode] By default (constants.RESTART_ON_REMOUNT) the saga will be started on component mount and
+ * cancelled with `task.cancel()` on component un-mount for improved performance. Another two options:
+ *   - constants.DAEMON—starts the saga on component mount and never cancels it or starts again,
+ *   - constants.ONCE_TILL_UNMOUNT—behaves like 'RESTART_ON_REMOUNT' but never runs it again.
+ *
+ */
+export default ({ key, saga, mode }) => (WrappedComponent) => {
```

Comment:
You have three options:
1. Start other sagas in a root saga in whatever way you see fit:

```js
export default function* rootSaga() {
    helloSaga(),
    watchIncrementAsync()
}
```
```js
yield all([
  takeLatest(LOAD_REPOS, getRepos),
  takeLatest(LOAD_USERS, getUsers),
]);
```

2. Decorate your component with `injectSaga` multiple times:

```js
export default compose(
  injectSaga({ key: 'home', homeSaga });
  injectSaga({ key: 'test', testSaga });
)(MyComponent);
```

3. Create your own decorator and use helpers from `utils/sagaInjectors`:

```js
injectors = getInjectors(this.context.store);
componentWillMount() {
  const { injectSaga } = this.injectors;
  sagaDescriptors.map(({ key, saga, mode }) => injectSaga(key, saga, this.props, mode));
}
```

The same is true for reducers, except step 1, to combine reducers into one you can use [`reduceReducers`](https://github.com/acdlite/reduce-reducers) or something like that.
If you have a convincing argument why these three options is not enough, we can change the API.